tests/integration/utils.go

```diff
@@ -272,6 +273,8 @@ func executeTestCase(
 		performAction(s, i, testCase.Actions[i])
 	}
 
+	resetMatchers(s)
```

Comment:
suggestion: A code comment documenting what you explained to Fred would be handy here, especially given that I'm hoping to remove the 'multiple test runs in a single test execution' and a comment would help encourage the removal of this call if/when that change gets made.